services/errors.js

```diff
@@ -1,34 +1,60 @@
 'use strict';
 
-class NotFound extends Error {
-  constructor(prettyMessage = 'not found') {
+class ShieldsRuntimeError extends Error {
+
+  get name() { return 'ShieldsRuntimeError'; }
+  get defaultPrettyMessage() { throw new Error('Must implement abstract method'); }
+
+  constructor(props, message) {
+    props = props || {};
+    super(message);
+    this.prettyMessage = props.prettyMessage || this.defaultPrettyMessage;
+    if (props.underlyingError) {
+      this.stack = props.underlyingError.stack;
+    }
+  }
+}
+
+class NotFound extends ShieldsRuntimeError {
+
+  get name() { return 'NotFound'; }
+  get defaultPrettyMessage() { return 'not found'; }
+
+  constructor(props) {
+    props = props || {};
```

Comment:
would we be better to remove this line and declare the default value in the header instead?
```javascript
constructor(props = {}) {
```

Reply:
:+1: updated